rts/System/Sound/OpenAL/Sound.cpp

```diff
@@ -529,6 +530,35 @@ void CSound::OpenLoopbackDevice(const std::string& deviceName)
 		return;
 	}
 
+	// 2.attempt: Workaround for SDL bug with some soundcards, returning more channels than we asked for (returned 8 in test case)
+	// (Maybe a third attempt should be taken into consideration: Disallow Channel Changes and ask for 1 channel only)
+	if (obtainedSpec.channels > 2) {
+		LOG("[Sound::%s] SDL returned %d channels, when we asked for %d. Closing previously opened device.", __func__, obtainedSpec.channels, desiredSpec.channels);
+		SDL_CloseAudioDevice(sdlDeviceID);
+		selectedDeviceName = "";
+		sdlDeviceID = 0;
+
+		if (!deviceName.empty()) {
+			LOG("[Sound::%s] opening configured device \"%s\" and disallow channel changes", __func__, deviceName.c_str());
+			sdlDeviceID = SDL_OpenAudioDevice(deviceName.c_str(), 0, &desiredSpec, &obtainedSpec, SDL_AUDIO_ALLOW_ANY_CHANGE & ~SDL_AUDIO_ALLOW_CHANNELS_CHANGE);
+			selectedDeviceName = deviceName;
+		}
+
+		if (sdlDeviceID == 0) {
+			LOG("[Sound::%s] opening default device and disallow channel changes", __func__);
+			sdlDeviceID = SDL_OpenAudioDevice(nullptr, 0, &desiredSpec, &obtainedSpec, SDL_AUDIO_ALLOW_ANY_CHANGE & ~SDL_AUDIO_ALLOW_CHANNELS_CHANGE);
+			selectedDeviceName = "default";
+		}
+
+		if (sdlDeviceID == 0) {
+			LOG("[Sound::%s] failed to open SDL audio, error:  \"%s\"", __func__, SDL_GetError());
+			SDL_QuitSubSystem(SDL_INIT_AUDIO);
+			return;
+		}
```

Comment:
a lot of this is copy-pasted from the existing code, perhaps extract to a common function?

Reply:
That's true. I stated this already in my comment on dc. Repeaded code is bad for maintenance and should not be used.
Seems ivand gave me the go in general, so i think i could that part too. 
Putting this to draft for now.